http-client/Network/HTTP/Client/Request.hs

```diff
@@ -127,8 +127,8 @@ applyAnyUriBasedAuth uri req =
   where
     hasAuth = (notEmpty theuser) && (notEmpty thepass)
     notEmpty = not . null
-    theuser = username authInfo
-    thepass = password authInfo
+    theuser = unEscapeString . takeWhile (/=':') . authPrefix $ authInfo
+    thepass = unEscapeString . takeWhile (/='@') . drop 1 . dropWhile (/=':') $ authInfo
```

Comment:
Why did you change these instead of modifying the implementation of `username` and `password`?